tools/tools.go

```diff
@@ -0,0 +1,7 @@
+// +build tools
```

Comment:
What is this file for?

Reply:
This creates a dependency on counterfeiter for the project. It has the effect that when you `go generate` to produce the code for the test fakes, each developer will use an identical version of counterfeiter as given in the module configuration. So it's an improvement on having counterfeiter as an external dependency where devs can have different versions producing difference code.